### DISCUSSION TITLE
How to create a friendly name/alias for an Orchard Core module

### DISCUSSION THREAD

**Discussion by mattobox** [OrchardCMS/OrchardCore#7219] [Q&A]
I understand that for Orchard Core project modules the default route can be referenced by the project name as the "area,"  however I was wondering if an alias could be created or if it can be changed to a friendlier name. Project names are often long and awkward. It would be nice if the module libraries and static files could map to this alias.

For the standard asp.net core files this seems to be a good option added to the startup.cs of the module library:
`
services.Configure<RazorPagesOptions>(options =>
			{
				options.Conventions.AddAreaFolderRoute("Module.Name", "/", "friendly");
			});
`
However, this doesn't work for the static files. Is there an Orchard Core implementation that provides the file contents of the application's module static view files. I suspect something like this in the startup.cs:
`
app.UseStaticFiles(new StaticFileOptions()
			{
				FileProvider = "An orchard core implementation with access to the static files",
				RequestPath = new PathString("/friendly")
			});
`

Any guidance or resources would be appreciated.

**Comment by ns8482e** [comment]
You can configure routes in your module startup see demo module https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Demo/Startup.cs

**Reply by mattobox** [reply]
Thanks, that’s a great resource. It doesn’t appear to have a solution for the static files within a module though. Any resource that shows how to do that?

**Reply by jtkech** [reply]
What you could do is to add, in your module / theme project file, this property

    <AssemblyName>Friendly</AssemblyName>

Then in your manifest file when defining your module / theme attribute

    [assembly: Theme(
        Id = "Friendly",

Unfortunately, in our msbuild .targets file i missed to replace `MSBuildProjectName` to `AssemblyName` in 2 places, maybe when i did this it was for a module without thinking about static asset files, i will do a PR soon to fix it.

Just fixed for testing and tried it with `TheTheme` and by using a new assembly name / id `Fiendly`, it works

![Friendly assets](https://user-images.githubusercontent.com/8586360/95400283-93449380-090a-11eb-90e1-b67d88c8c208.png)

**Reply by mattobox** [reply]
Thanks for your response. Yes, I tried adding those "Friendly" references and it didn't work. So, am I to understand that this change will be in a future release of the project?

**Reply by jtkech** [reply]
I will fix it soon in the dev branch, so you would need to use the source code or to reference the packages from the cloudsmith feed. Otherwise, you will have to wait for the future public release on Nuget that is in preparation #7206
